hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/RegionNormalizerStateStore.java

```diff
@@ -15,60 +15,52 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package org.apache.hadoop.hbase.zookeeper;
+package org.apache.hadoop.hbase.master.normalizer;
 
 import java.io.IOException;
-import org.apache.hadoop.hbase.Abortable;
 import org.apache.hadoop.hbase.exceptions.DeserializationException;
-import org.apache.hadoop.hbase.util.Bytes;
+import org.apache.hadoop.hbase.master.MasterStateStore;
+import org.apache.hadoop.hbase.master.region.MasterRegion;
+import org.apache.hadoop.hbase.zookeeper.ZKWatcher;
 import org.apache.yetus.audience.InterfaceAudience;
 import org.apache.zookeeper.KeeperException;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 import org.apache.hadoop.hbase.shaded.protobuf.ProtobufUtil;
 import org.apache.hadoop.hbase.shaded.protobuf.generated.RegionNormalizerProtos;
 
 /**
- * Tracks region normalizer state up in ZK
+ * Store region normalizer state.
  */
 @InterfaceAudience.Private
-public class RegionNormalizerTracker extends ZKNodeTracker {
-  private static final Logger LOG = LoggerFactory.getLogger(RegionNormalizerTracker.class);
+public class RegionNormalizerStateStore extends MasterStateStore {
```

Comment:
this also looks like it should use SwitchStateStore? would be nice to unify all these switches. just needs an abstract toByteArray/parseFrom

Reply:
Maybe it is OK since these classes are all IA.Private so we are free to change their methods. Let me have a try.

Reply:
Done.